lib/govuk_schemas/random_example.rb

```diff
@@ -34,41 +34,23 @@ def self.for_schema(schema_key_value)
       GovukSchemas::RandomExample.new(schema: schema)
     end
 
-    # Return a hash with a random content item
-    #
-    # Example:
-    #
-    #     GovukSchemas::RandomExample.for_schema("detailed_guide", schema_type: "frontend").payload
-    #     # => {"base_path"=>"/e42dd28e", "title"=>"dolor est...", "publishing_app"=>"elit"...}
-    #
-    # @return [Hash] A content item
-    def payload
-      item = @random_generator.payload
-      errors = validation_errors_for(item)
-
-      if errors.any?
-        raise InvalidContentGenerated, error_message(item, errors)
-      end
-
-      item
-    end
-
-    # Return a content item merged with a hash. If the resulting content item
-    # isn't valid against the schema an error will be raised.
+    # Return a content item merged with a hash and with the excluded fields removed.
+    # If the resulting content item isn't valid against the schema an error will be raised.
     #
     # Example:
     #
     #      random = GovukSchemas::RandomExample.for_schema("detailed_guide", schema_type: "frontend")
-    #      random.merge_and_validate(base_path: "/foo")
+    #      random.customise_and_validate({base_path: "/foo"}, ["withdrawn_notice"])
     #      # => {"base_path"=>"/e42dd28e", "title"=>"dolor est...", "publishing_app"=>"elit"...}
     #
     # @param [Hash] hash The hash to merge the random content with
+    # @param [Array] array The array containing fields to exclude
     # @return [Hash] A content item
     # @raise [GovukSchemas::InvalidContentGenerated]
-    def merge_and_validate(user_defined_values)
+    def customise_and_validate(user_defined_values = {}, fields_to_exclude = [])
```

Comment:
At the moment it is not. Do you happen to know of a use case where this would be necessary currently?

We had a think about how to implement that and what the interface might look like. Aside from recursively checking all fields in the hash, this could be an example:

`customise_and_validate({}, ['schema_name', ['details', 'description'], ['details', 'name']])`

As the interface currently stands we feel that what we have done will be useful, but could also be expanded to nested fields in the future.